docs/advanced-features/good-practice.md

```diff
@@ -101,5 +103,767 @@ If enabled (see set-up instructions above), these events will be
+    "isActive": true,
+    "isSsoOnly": false,
+    "isService": false,
+    "dateJoined": "2020-11-16T03:32:01Z",
+    "roleId": "*REDACTED*"
+  },
+  "requestParameters": { "username": "example_user", "password": "***" },
+  "responseElements": {
+    "access_token": "***",
+    "refresh_token": "***",
+    "exp": "2023-09-06T13:59:36Z"
+  },
+  "errorCode": null,
+  "errorMessage": null,
+  "additionalEventData": { "method": "password" }
+}
+```
+
+A service user (Canary) authenticated:
+
+```json
+{
+  "eventVersion": 1,
+  "eventRevision": 0,
+  "eventTime": "2023-06-12T00:37:21Z",
+  "eventID": "10b0901a-74cd-4e62-8954-33f4fba6e8ad",
+  "eventSource": "QuiltServer",
+  "type": "QuiltApiCall",
+  "eventName": "Auth.ServiceLogin",
+  "userAgent": "",
+  "sourceIPAddress": "*REDACTED*",
+  "userIdentity": {
+    "type": "QuiltUser",
+    "id": "*REDACTED*",
+    "userName": "_canary",
+    "email": "canary@quiltdata.io",
+    "isAdmin": false,
+    "lastLogin": "2022-11-24T13:55:31Z",
+    "isActive": true,
+    "isSsoOnly": false,
+    "isService": true,
+    "dateJoined": "2022-11-24T13:55:31Z",
+    "roleId": "*REDACTED*"
+  },
+  "requestParameters": { "provider": "quilt-service-auth", "token": "***" },
+  "responseElements": {
+    "access_token": "***",
+    "refresh_token": "***",
+    "exp": "2023-09-10T00:37:21Z"
+  },
+  "errorCode": null,
+  "errorMessage": null,
+  "additionalEventData": { "account_id": "_canary" }
+}
+```
+
+### Querying With Athena
+
+Audit events can be queried with [AWS Athena](https://aws.amazon.com/athena/).
+Quilt Stack provisions the following resources:
+
+- **Audit Trail Database**: a Glue database named `audittraildatabase-${random_string}`,
+  exposed as `AuditTrailDatabase` stack resource.
+
+- **Audit Trail Table**: a Glue table named `audit_trail` in **Audit Trail Database**.
+
+- **Audit Workgroup**: an Athena workgroup named `${AWS::StackName}-audit"`,
+  exposed as `AuditTrailWorkgroup` stack resource.
+
+- **Audit Trail Bucket**: an S3 bucket storing all the audit trail data and
+  Athena query results for **Audit Workgroup**.
+
+In order to query audit trail data via AWS Athena Console, you should:
+
+1. Select `AwsDataCatalog` from the "Data source" dropdown.
+
+2. Select **Audit Trail Database** from the "Database" dropdown.
+
+3. Select **Audit Workgroup** from the "Workgroup" dropdown.
+
+All the events are available in the `audit_trail` table,
+which has the following fields (schema version 1.0):
+
+- `eventversion: tinyint`
+- `eventrevision: tinyint`
+- `eventtime: timestamp`
+- `eventid: string`
+- `eventsource: string`
+- `eventtype: string`
+- `eventname: string`
+- `useragent: string`
+- `sourceipaddress: string`
+- `useridentity: string`
+- `requestparameters: string`
+- `responseelements: string`
+- `errorcode: string`
+- `errormessage: string`
+- `additionaleventdata: string`
+- `date: string` (partition)
+
+The data is partitioned by `date`, which has `YYYY/mm/dd` format.
```

Comment:
Probably we should add link to doc that explains what is partitioning (e.g. https://docs.aws.amazon.com/athena/latest/ug/ctas-partitioning-and-bucketing.html#ctas-partitioning-and-bucketing-what-is-partitioning) and suggest to always use partitioning to pay less many.